### PR TITLE
[WIP] Refactoring

### PR DESCRIPTION
**Pull request by matej21**

### CODE REVIEW

File: src/Mapper/Dbal/DbalMapper.php [hunk 1]
```diff
@@ -37,15 +37,16 @@ class DbalMapper extends BaseMapper
...
+	private $mapperCoordinator;
 
 
-	public function __construct(Connection $connection, Cache $cache)
+	public function __construct(Connection $connection, DbalMapperCoordinator $mapperCoordinator, Cache $cache)
 	{
 		$key = md5(json_encode($connection->getConfig()));
 		$this->connection = $connection;
 		$this->cache = $cache->derive('mapper.' . $key);
+		$this->mapperCoordinator = $mapperCoordinator;
```
Comment (hrach) [inline src/Mapper/Dbal/DbalMapper.php]: plese move the assign one line above to match the arguments :-)

File: src/Mapper/Dbal/DbalMapperCoordinator.php [hunk 2]
```diff
@@ -0,0 +1,50 @@
+<?php declare(strict_types = 1);
+
+namespace Nextras\Orm\Mapper\Dbal;
+
+use Nextras\Dbal\Connection;
+
+
+class DbalMapperCoordinator
+{
+
```
Comment (hrach) [inline src/Mapper/Dbal/DbalMapperCoordinator.php]: drop the line

File: src/Mapper/Dbal/DbalMapperCoordinator.php [hunk 3]
```diff
@@ -0,0 +1,50 @@
+<?php declare(strict_types = 1);
+
```
Comment (hrach) [inline src/Mapper/Dbal/DbalMapperCoordinator.php]: missing phpdoc

File: src/Mapper/Dbal/DbalMapperCoordinator.php [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+
+	public function rollback()
+	{
+		if ($this->transactionActive) {
+			$this->connection->rollbackTransaction();
+			$this->transactionActive = false;
+		}
+	}
+
```
Comment (hrach) [inline src/Mapper/Dbal/DbalMapperCoordinator.php]: drop the empty line

File: src/Mapper/Dbal/StorageReflection/StorageReflection.php [hunk 5]
```diff
@@ -15,7 +15,7 @@
 use Nextras\Dbal\Platforms\IPlatform;
 use Nextras\Orm\InvalidArgumentException;
 use Nextras\Orm\InvalidStateException;
-use Nextras\Orm\Mapper\IMapper;
+use Nextras\Orm;
```
Comment (hrach) [inline src/Mapper/Dbal/StorageReflection/StorageReflection.php]: is it correctly sorted?

File: src/Repository/IdentityMap.php [hunk 6]
```diff
@@ -24,23 +24,12 @@ class IdentityMap extends Object
...
-
 	/** @var ReflectionClass[] */
 	private $entityReflections;
 
-	/** @var IDependencyProvider */
-	private $dependencyProvider;
-
-
-	public function __construct(IRepository $repository, IDependencyProvider $dependencyProvider = null)
+	public function __construct(IRepository $repository)
```
Comment (hrach) [inline src/Repository/IdentityMap.php]: empty line before consturctor

File: src/Mapper/IMapper.php [hunk 7]
```diff
@@ -60,12 +53,6 @@ public function setRepository(IRepository $repository);
 	public function getRepository(): IRepository;
 
 
-	public function getTableName(): string;
```
Comment (hrach) [inline src/Mapper/IMapper.php]: why is this method removed?

Comment (hrach) [inline src/Mapper/IMapper.php]: but probably ok :)

Comment (matej21) [inline src/Mapper/IMapper.php]: it is only used by a mapper

**Comment by matej21** [comment]
@hrach done

**Comment by hrach** [comment]
🎉 
thanks!

**Pull request merged**